### ISSUE TITLE
USB and Ethernet ports from a type-c dongle not recognised

### ISSUE THREAD

**Issue by freeyoung** [tylernguyen/x1c6-hackintosh#104]
First of all, many thanks! I've managed to upgrade from Clover + Mojave to OpenCore + Big Sur in the past few days.

After the upgrade, almost everything has been working great, except for my type-c dongle.

The dongle has 3x USB type-A, 1x ethernet, 1x HDMI and PD support. HDMI and PD have no problems at all, but I can not see the USB 3.0 Hub in the USB section in System Information. I guess that the absence of the ethernet port is also caused by that.

I followed the instruction [here](https://www.tonymacx86.com/threads/macos-catalina-10-15-0-usb-port-limit-removal-patch.285098/), and the type-A ports worked, the ethernet was green in Network, but it could not get an IPv4 from DHCP. So I think maybe the "fix" above was not the right way to go?

BTW, the dongle works perfectly fine when I connect it to my MacBook Air.

**Comment by benbender** [comment]
@freeyoung the instructions are quite outdated and also don't fit an usb-c-dongle.

As the USB-C-controller is part of the Thunderbolt (TB) controller, please check in BIOS in which mode you are running TB ("Windows 10" or "Bios Assist"). Additionally I would ask you to save and upload an IOReg-dump (https://github.com/vulgo/IORegistryExplorer, use "Save as") when the dongle is attached. After that, we may be able to identify the problem.

**Comment by freeyoung** [comment]
@benbender Thanks for the hint! I followed the TB3 instruction in https://github.com/tylernguyen/x1c6-hackintosh/blob/master/docs/1_README-HARDWAREandBIOS.md and chose the second option:

> Thunderbolt BIOS Assist Mode: Disabled
> Thunderbolt(TM) Device: Disabled

Now I can see the USB 3.0 Hub and the ethernet device `USB 10/100/1000 LAN`. However, it still could not obtain an IP address from DHCP, maybe due to something else.

I've made an IOReg-dump when the dongle is attached:

[midgard.zip](https://github.com/tylernguyen/x1c6-hackintosh/files/5550352/midgard.zip)

**Comment by Porco-Rosso** [comment]
Just to chime in, I've faced the exact same situation as @freeyoung

**Comment by benbender** [comment]
The "not getting an IP"-problem is a bit out of topic for this repo. If the driver loads correctly, the basic setup works. The actual problem could be everything from bad drivers, cabling through DHCP-problems. To rule most of the problems out, I would suggest to try the cable/connection on a different machine to try to pin down the problem.

**Comment by freeyoung** [comment]
> The "not getting an IP"-problem is a bit out of topic for this repo. If the driver loads correctly, the basic setup works. The actual problem could be everything from bad drivers, cabling through DHCP-problems. To rule most of the problems out, I would suggest to try the cable/connection on a different machine to try to pin down the problem.

Thanks for your response. I agree that the DHCP problem could be a bit unrelated.

To provide more info, the ethernet port is using RTL8153 and its driver is from macOS. I tried multiple times to unplug the dongle from x1c6 to a macbook air running Big Sur 11.0.1, the ethernet works fine and can get an IPv4 via DHCP in a few seconds.

**Comment by freeyoung** [comment]
> Can you get an IP using manual settings?

@Porco-Rosso unfortunately not, I set the IPv4 manually (IP + Netmask + Gateway + DNS) but could not even ping the gateway IP 😢

**Comment by benbender** [comment]
@freeyoung Realtek-drivers are flaky in OSX as parts of it was/is(?) done in userspace. Had massive problems with a similar device in Catalina where the native driver ate up 100% CPU. Only tip I have at hand is to check the logs (`log show --last 5m`/`sudo dmesg`) - apart of that, I think this issue can be closed here.

**Comment by freeyoung** [comment]
I reviewed the logs but there was nothing interesting. Also, I learned from 10x reboots that the ethernet was only able to be recognized for a chance at around 60% in the meantime the USB 3.0 Hub was always there.

I'd give up on the dongle completely, and turn to the official mini-RJ45 to RJ45 adapter. Considering this together with https://github.com/tylernguyen/x1c6-hackintosh/issues/91, I might buy something like https://www.amazon.de/dp/B078JZMR9L/ for PD and a stable HDMI only.

This issue can be closed.
